sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala

```diff
@@ -31,7 +31,7 @@ import org.apache.spark.sql.internal.SQLConf
  * input partition ordering requirements are met.
  */
 case class EnsureRequirements(conf: SQLConf) extends Rule[SparkPlan] {
-  private def defaultNumPreShufflePartitions: Int = conf.numShufflePartitions
+  private lazy val defaultNumPreShufflePartitions: Int = conf.numShufflePartitions
```

Comment:
Lazy val means that any updates to conf would  not reflect in this class. Given that this class is created in queryexecution, I can't think of a case where it could create a problem. However, any specific reason for making this a lazy val?